scripts/zkai_d128_gate_value_compact_preprocessed_gate.py

```
@@ -0,0 +1,626 @@
+        raise GateValueCompactGateError("evidence relative path drift")
+    if require_exact_int(row.get("proof_json_size_bytes"), "proof JSON bytes") != proof_json_size:
+        raise GateValueCompactGateError("proof JSON byte drift")
+    accounting = require_dict(row.get("local_binary_accounting"), "local binary accounting")
+    if require_exact_int(accounting.get("typed_size_estimate_bytes"), "typed size") != local_typed_bytes:
+        raise GateValueCompactGateError("typed size drift")
+    if require_exact_int(accounting.get("component_sum_bytes"), "component sum") != local_typed_bytes:
+        raise GateValueCompactGateError("component sum drift")
+    records = require_list(accounting.get("records"), "records")
+    counts = {require_dict(record, "record").get("path"): require_exact_int(require_dict(record, "record").get("item_count"), "record count") for record in records}
+    if counts != expected_counts:
+        raise GateValueCompactGateError("record counts drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)